src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -47,10 +50,29 @@
     <DebugEngines>{2E36F1D4-B23C-435D-AB41-18E608940038}</DebugEngines>
   </PropertyGroup>
 
+  <!--
+    combine the TestTargetFramework with IterationCount.  by batching on TestTargetFrameworkBatch
+    in the RunTestsForProject Outputs it effectively allows us to run the target multiple times,
+    once for each IterationCount.  if IterationCount isn't defined there is no effect.  right now
+    this is used for perf testing which requires us to run a test multiple iterations and aggregate
+    the results.
+  -->
+  <Target Name="InitIterationBatch"
+          BeforeTargets="RunTestsForProject">
+
+    <ItemGroup>
+      <TestTargetFrameworkBatch Include="@(TestTargetFramework)">
+        <Count>%(IterationCount.Identity)</Count>
+      </TestTargetFrameworkBatch>
+    </ItemGroup>
+
+  </Target>
+
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
           AfterTargets="CheckTestCategories"
-          Condition="'$(TestDisabled)'!='true'">
+          Condition="'$(TestDisabled)'!='true'"
+          Outputs="%(TestTargetFrameworkBatch.Folder) %(TestTargetFrameworkBatch.Count)">
```

Comment:
You are going to clash with @eatdrinksleepcode current PR https://github.com/dotnet/buildtools/pull/116